auth/src/main/java/com/firebase/ui/auth/util/data/AuthViewModelBase.java

```diff
@@ -0,0 +1,40 @@
+package com.firebase.ui.auth.util.data;
+
+import android.app.Application;
+
+import com.firebase.ui.auth.util.ui.FlowHolder;
+import com.firebase.ui.auth.util.ui.ViewModelBase;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.PhoneAuthProvider;
+
+public class AuthViewModelBase extends ViewModelBase<FlowHolder> {
+    private FlowHolder mFlowHolder;
+    private FirebaseAuth mAuth;
+    private PhoneAuthProvider mPhoneAuth;
+
+    protected AuthViewModelBase(Application application) {
+        super(application);
+    }
+
+    @Override
+    protected void onCreate(FlowHolder args) {
```

Comment:
Similar to my previous comment, seems like we could pass `FlowParameters` here, right?  We only use the `FlowHolder` in `getFlowHolder()` but that is also defined by the base `Activity` class.